code/_onclick/item_attack.dm

```diff
@@ -67,7 +67,7 @@
 		else
 			SStutorials.suggest_tutorial(user, /datum/tutorial/drop, params2list(params))
 
-	return afterattack(target, user, TRUE, params) == TRUE
+	return afterattack_result == TRUE
```

Comment:
check for truthyness here, not just true. afterattack can return AFTERATTACK_PROCESSED_ITEM now which is 2

Reply:
```suggestion
	return afterattack_result
```

Reply:
I don't understand this code is new to me. What type of checks do u mean exactly?

Reply:
I'll try something first

Reply:
Dont' do this

Reply:
It intentionally does not check for truthiness because truthiness means nothing here. The flag is just a flag that is often mass applied for the sake of convenience, and TRUE/FALSE results are inconsistent and useless but intentionally preserved here for the sake of preserving whatever old bullshit behavior was expecting them